tests/const_/mod.rs

```diff
@@ -111,3 +124,124 @@ fn const_in_scope() {
 
     assert_eq!(actual.out, "x");
 }
+
+#[test]
+fn complex_const_export() {
+    let inp = &[MODULE_SETUP, "use spam", "$spam.X"];
+    let actual = nu!(&inp.join("; "));
+    assert_eq!(actual.out, "x");
+
+    let inp = &[MODULE_SETUP, "use spam", "$spam.eggs.E"];
+    let actual = nu!(&inp.join("; "));
+    assert_eq!(actual.out, "e");
+
+    let inp = &[MODULE_SETUP, "use spam", "$spam.eggs.bacon.viking"];
+    let actual = nu!(&inp.join("; "));
+    assert_eq!(actual.out, "eats");
+
+    let inp = &[
+        MODULE_SETUP,
+        "use spam",
+        "($spam.eggs.bacon.none | is-empty)",
+    ];
+    let actual = nu!(&inp.join("; "));
+    assert_eq!(actual.out, "true");
+}
+
+#[test]
+fn complex_const_glob_export() {
+    let inp = &[MODULE_SETUP, "use spam *", "$X"];
+    let actual = nu!(&inp.join("; "));
+    assert_eq!(actual.out, "x");
+
+    let inp = &[MODULE_SETUP, "use spam *", "$eggs.E"];
+    let actual = nu!(&inp.join("; "));
+    assert_eq!(actual.out, "e");
+
+    let inp = &[MODULE_SETUP, "use spam *", "$eggs.bacon.viking"];
+    let actual = nu!(&inp.join("; "));
+    assert_eq!(actual.out, "eats");
+
+    let inp = &[MODULE_SETUP, "use spam *", "($eggs.bacon.none | is-empty)"];
+    let actual = nu!(&inp.join("; "));
+    assert_eq!(actual.out, "true");
+}
+
+#[test]
+fn complex_const_drill_export() {
+    let inp = &[
+        MODULE_SETUP,
+        "use spam eggs bacon none",
+        "($none | is-empty)",
+    ];
+    let actual = nu!(&inp.join("; "));
+    assert_eq!(actual.out, "true");
+}
+
+#[test]
+fn complex_const_list_export() {
+    let inp = &[
+        MODULE_SETUP,
+        "use spam [X eggs]",
+        "[$X $eggs.E] | str join ''",
+    ];
+    let actual = nu!(&inp.join("; "));
+    assert_eq!(actual.out, "xe");
+}
+
+#[test]
+fn exported_const_is_const() {
+    let module1 = "module foo {
+        export def main [] { 'foo' }
+    }";
+
+    let module2 = "module spam {
+        export const MOD_NAME = 'foo'
+    }";
+
+    let inp = &[module1, module2, "use spam", "use $spam.MOD_NAME", "foo"];
```

Comment:
if i'm not mistaken
```rust
&[module1, module2, "use spam", "use $spam.MOD_NAME", "foo"].join("\n")
```
will translate to
```nushell
module foo {
    export def main [] { 'foo' }
}
module spam {
    export const MOD_NAME = 'foo'
}
use spam
use $spam.MOD_NAME
foo
```
but it appears to me this is trying to call the `main` command from the `foo` module which was not `use`d :confused: 

### it looks to me this should return an error, right?

Reply:
It's `use`d implicitly when you call `use foo`.

Reply:
do you mean `use spam`? because i do not see a `use foo` :confused:

Reply:
`use $spam.MOD_NAME` translates to `use foo`

Reply:
oooooh

Reply:
i get it

Reply:
need to get used to `use` being able to take constants like that :crossed_fingers: